packages/common/src/types/ccdAmount.ts

```diff
@@ -22,7 +49,83 @@ export class CcdAmount {
         this.microCcdAmount = microCcdAmount;
     }
 
+    /**
+     * Returns the amount of micro CCD as a Big.
+     *
+     * @returns The amount of micro CCD as a Big
+     */
+    toBig(): Big {
+        return Big(this.microCcdAmount.toString());
+    }
+
+    /**
+     * Returns the amount of CCD as a Big.
+     *
+     * @returns The amount of CCD as a Big
+     */
+    toCcd(): Big {
+        return this.toBig().div(Big(MICRO_CCD_PER_CCD));
+    }
+
+    /**
+     * Creates a CcdAmount from a number, string, big or bigint.
+     *
+     * @param ccd The amount of CCD
+     * @returns The CcdAmount object derived from the ccd input parameter
+     * @throws If the number of commas in the input string exceeds 1
+     * @throws If a number is passed with several decimal seperators
+     * @throws If a negative amount of micro CCD is passed
+     * @throws If the micro CCD passed is greater than what can be contained in a 64-bit integer
+     */
+    static fromCcd(ccd: BigSource | bigint): CcdAmount {
+        if (typeof ccd === 'bigint') {
+            ccd = ccd.toString();
+        }
+
+        const microCcd = newBig(ccd).mul(Big(MICRO_CCD_PER_CCD));
+        return new CcdAmount(microCcd);
+    }
+
+    /**
+     * Converts an amount of CCD to micro CCD and asserts that the amount is a valid amount of CCD.
+     *
+     * @param ccd The amount of CCD
+     * @returns The amount of micro CCD as a Big
+     * @throws If the number of commas in the input string exceeds 1
+     * @throws If a number is passed with several decimal seperators
```

Comment:
* @throws If a negative amount of micro CCD is passed
 * @throws If the micro CCD passed is greater than what can be contained in a 64-bit integer